src/packages/recompose/createElement.js

```diff
@@ -2,12 +2,12 @@ import React from 'react'
 import isReferentiallyTransparentFunctionComponent
   from './isReferentiallyTransparentFunctionComponent'
 
-const createElement = (Component, props, children) => {
+const _createElement = (isReferentiallyTransparent, Component, props, children) => {
   /* eslint-disable */
   const hasKey = props && props.hasOwnProperty('key')
   /* eslint-enable */
 
-  if (!hasKey && isReferentiallyTransparentFunctionComponent(Component)) {
+  if (!hasKey && isReferentiallyTransparent) {
```

Comment:
for internal use there's no point in checking `hasKey` since react removes `key` from props before passing them down to the component https://github.com/facebook/react/blob/master/src/isomorphic/classic/element/ReactElement.js#L149)